src/Stratis.Features.FederatedPeg/Wallet/TransactionData.cs

```diff
@@ -74,14 +104,18 @@ public Money SpendableAmount(bool confirmedOnly)
             // This method only returns a UTXO that has no spending output.
             // If a spending output exists (even if its not confirmed) this will return as zero balance.
             if (!this.IsSpendable()) return Money.Zero;
-            
+
             if (confirmedOnly && !this.IsConfirmed())
             {
                 return Money.Zero;
             }
 
             return this.Amount;
+        }
 
+        public void Subscribe(ITransactionDataObserver observer)
```

Comment:
Can you describe more about why we need this? It's so all instances of TransactionData are updated when one is changed right? And avoids cloning?

Given that we use locks I'm not sure we need this, seems to be adding opacity

Reply:
It used to set the observer so that the observer can be notified if fields on the observed object are changed - e.g. SpendingDetails, This allows the observer to update its spendable transactions lookup.